core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/PoolQC.java

```diff
@@ -12,61 +12,51 @@
  *
  * MISO is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public License
- * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ * along with MISO. If not, see <http://www.gnu.org/licenses/>.
  *
  * *********************************************************************
  */
 
 package uk.ac.bbsrc.tgac.miso.core.data;
 
-//import com.fasterxml.jackson.annotation.*;
-//import com.fasterxml.jackson.databind.annotation.JsonSerialize;
-import org.codehaus.jackson.annotate.JsonBackReference;
-import org.codehaus.jackson.annotate.JsonIgnoreProperties;
-import org.codehaus.jackson.annotate.JsonTypeInfo;
-import org.codehaus.jackson.map.annotate.JsonSerialize;
-import uk.ac.bbsrc.tgac.miso.core.exception.MalformedPoolException;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.PoolImpl;
 
 /**
- * A QC that is specifically carried out on a given {@link uk.ac.bbsrc.tgac.miso.core.data.Pool}
- *
+ * Concrete implementation of a PoolQC
+ * 
  * @author Rob Davey
  * @since 0.1.9
  */
-@JsonSerialize(typing = JsonSerialize.Typing.STATIC)
-@JsonTypeInfo(use=JsonTypeInfo.Id.CLASS, include= JsonTypeInfo.As.PROPERTY, property="@class")
-public interface PoolQC extends QC {
-  /**
-   * Returns the pool of this PoolQC object.
-   *
-   * @return Pool pool.
-   */
-  @JsonBackReference(value = "qcpool")
-  public Pool getPool();
+@Entity
+@Table(name = "PoolQC")
+public class PoolQC extends QC {
+  private static final long serialVersionUID = 1L;
+
+  @ManyToOne(targetEntity = PoolImpl.class)
+  @JoinColumn(name = "pool_poolId")
+  private Pool pool;
+
+
+  @Override
+  public QualityControllable<?> getEntity() {
+    return pool;
+  }
 
-  /**
-   * Sets the pool of this PoolQC object.
-   *
-   * @param pool Pool.
-   * @throws uk.ac.bbsrc.tgac.miso.core.exception.MalformedPoolException when the Pool being set is not valid
-   */
-  public void setPool(Pool pool) throws MalformedPoolException;
+  public Pool getPool() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "getEntity" on line 50. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)